File: app/src/main/java/com/wosloveslife/fantasy/interfaces/IPlay.java
package com.wosloveslife.fantasy.interfaces;

import android.support.annotation.IntDef;

import com.wosloveslife.dao.Audio;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static com.wosloveslife.fantasy.interfaces.IPlay.PlayState.PLAY_STATE_COMPLETE;
import static com.wosloveslife.fantasy.interfaces.IPlay.PlayState.PLAY_STATE_IDLE;
import static com.wosloveslife.fantasy.interfaces.IPlay.PlayState.PLAY_STATE_PAUSE;
import static com.wosloveslife.fantasy.interfaces.IPlay.PlayState.PLAY_STATE_PLAY;

/**
 * Created by zhangh on 2017/1/2.
 */
public interface IPlay {

    @IntDef({PLAY_STATE_IDLE, PLAY_STATE_PLAY, PLAY_STATE_PAUSE, PLAY_STATE_COMPLETE})
    @Retention(RetentionPolicy.SOURCE)
    @interface PlayState {
        int PLAY_STATE_IDLE = 0;
        int PLAY_STATE_PLAY = 1;
        int PLAY_STATE_PAUSE = 2;
        int PLAY_STATE_COMPLETE = 3;
    }

    void play(Audio audio);

    void pause();

    void seekTo(long progress);
}

File: player/src/main/java/com/wosloveslife/player/IPlay.kt
package com.wosloveslife.player

import android.support.annotation.IntDef

import com.google.android.exoplayer2.ExoPlayer


/**
 * Created by zhangh on 2017/1/2.
 */
interface IPlay {

    @IntDef(PLAY_STATE_IDLE, PLAY_STATE_PLAY, PLAY_STATE_PAUSE, PLAY_STATE_COMPLETE)
    @Retention(AnnotationRetention.SOURCE)
    annotation class PlayState

    companion object {
        const val PLAY_STATE_IDLE = 0L
        const val PLAY_STATE_PLAY = 1L
        const val PLAY_STATE_PAUSE = 2L
        const val PLAY_STATE_COMPLETE = 3L
    }

    fun play(audio: AudioResource)

    fun pause()

    fun seekTo(progress: Long)

    fun isPlaying() : Boolean

    fun addListener(listener: ExoPlayer.EventListener)
}
